tests/test_rand_weighted_cropd.py

```diff
@@ -139,6 +139,24 @@ def test_rand_weighted_crop_bad_w(self):
         np.testing.assert_allclose(result[0]["seg"].shape, (1, 48, 64, 80))
         np.testing.assert_allclose(np.asarray(crop.centers), [[24, 32, 40], [24, 32, 40], [24, 32, 40]])
 
+    def test_rand_weighted_crop_patch_index(self):
+        img = self.imt[0]
+        n_samples = 3
+        crop = RandWeightedCropd(("img", "seg"), "w", (10, -1, -1), n_samples)
+        weight = np.zeros_like(img)
+        weight[0, 7, 17] = 1.1
+        weight[0, 13, 31] = 1.1
+        weight[0, 24, 21] = 1
+        crop.set_random_state(10)
+        result = crop({"img": img, "seg": self.segn[0], "w": weight, "img_meta_dict": {"affine": None}})
+        self.assertTrue(len(result) == n_samples)
+        np.testing.assert_allclose(np.asarray(crop.centers), [[14, 32, 40], [41, 32, 40], [20, 32, 40]])
+        for i in range(n_samples):
```

Comment:
should the unit test not try to modify some metadata of one of the samples, and then verifying that it hasn't changed for the other samples? (check copy was deep, not shallow)

Reply:
Hi @rijobro ,

I already added the `"img_meta_dict": {"affine": None}` in the test data, if shallow copy, all the samples from `img` will have the same patch index.

Thanks.